server/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,46 @@
+import { Request, Response, NextFunction } from "express";
+import { User } from "../interfaces/user.interface";
+import jwt from "jsonwebtoken";
+import mysql, { RowDataPacket } from "mysql2/promise";
+
+export const protect = async function (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  let token: string;
+
+  if (
+    req.headers.authorization &&
+    req.headers.authorization.startsWith("Bearer")
+  ) {
+    try {
+      token = req.headers.authorization.split(" ")[1];
+
+      const decoded = jwt.verify(token, process.env.JWT_SECRET!) as {
+        id: string;
+      };
+
+      // query here
+      const connection = await mysql.createConnection(
+        process.env.DATABASE_URL as string
+      );
+      const query = `SELECT * FROM users WHERE id = ?`;
+
+      const [rows] = await connection.execute<RowDataPacket[]>(query, [
+        decoded.id,
+      ]);
+
+      const foundUser: User = rows[0] as User;
```

Comment:
This query isn't necessary, any table that references data belonging to this user should be accessible by user id which can be pulled and verified as legitimate from the token and jwt.verify. In theory that saves a database call on every protected route call, curious as to what your thought process was here though if you think the other user information might be useful generically on protected requests.

Reply:
Yeah, it's probably nitpicky and the end result is the same.  As long as the user id is available on the request via the token decode middleware it shouldnt be an issue regardless of it being the full user payload or not.  Just a minor optimization if we dont need the other user information on authenticated requests